core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -54,6 +56,7 @@
  * 
  * @since 2.0
  */
+@Restricted(NoExternalUse.class) // use by Jelly
 public class SetupWizard {
```

Comment:
> In the future plugin instances may hook on the wizard during the installation process

Not sure how that would work. If and when we have such a requirement, an API can be designed accordingly.